core/domain/suggestion_registry.py

```diff
@@ -306,11 +306,12 @@ def populate_old_value_of_change(self) -> None:
             'Subclasses of BaseSuggestion should implement '
             'populate_old_value_of_change.')
 
-    # TODO(#16047): Here we use type Any because BaseSuggestion class is not
-    # implemented according to the strict typing which forces us to use Any
-    # here so that MyPy does not throw errors for different types of values
-    # used in sub-classes. Once this BaseSuggestion is refactored, we can
-    # remove type Any from here.
+    # TODO(#16047): Here we use type Any because the method pre_update_validate
+    # is used inside sub-classes with different argument types, which according
+    # to MyPy violates the 'Liskov substitution principle' and throws an error
+    # in every sub-class where this pre_update_validate method is used, so to
```

Comment:
...used. So, to...

Also, when you say (below) "once this BaseSuggestion class is refactored", what do you mean? You should give the exact condition for when to remove type Any, and there are other refactorings of BaseSuggestion that are possible that wouldn't solve this issue.

Reply:
> ..used. So, to...

Done

> You should give the exact condition for when to remove type Any, and there are other refactorings of BaseSuggestion that are possible that wouldn't solve this issue

I'll add more info on the issue itself, as it requires more investigation!